cmake/IgnUtils.cmake

```diff
@@ -1804,3 +1804,366 @@ macro(_ign_cmake_parse_arguments prefix options oneValueArgs multiValueArgs)
   endif()
 
 endmacro()
+ 
+#####GSOC##########################################
+
+#ign_add_resources(path_to_resources)
+#Installs the folder and provides it's path to ign_environment_hook()
+macro(ign_add_resources path_to_resources)
+
+  install(DIRECTORY
+  ${path_to_resources}
+  DESTINATION share/${PROJECT_NAME})
+
+  list(APPEND resources_path ${path_to_resources})
+
+endmacro() 
+#####################################################
+#ign_environment_hook()
+#Get's path from other macros and creates hooks to export them
+#Currently it creates colcon.pkg along with hooks.dsv for exporting them
+#Mechanisms for plain cmake packages is under development 
+macro(ign_environment_hook export_type)
+  list(REMOVE_DUPLICATES resources_path)
+  list(REMOVE_DUPLICATES plugins_path)
+  file(WRITE ${CMAKE_CURRENT_SOURCE_DIR}/hooks/hook.dsv.in
+    "")
+  if(${export_type} STREQUAL "colcon")
+
+    foreach(resource_path ${resources_path})
+      file(APPEND ${CMAKE_CURRENT_SOURCE_DIR}/hooks/hook.dsv.in
+      "prepend-non-duplicate;IGN_GAZEBO_RESOURCE_PATH;@CMAKE_INSTALL_PREFIX@/share/@PROJECT_NAME@/${resource_path}\n")
+    endforeach()
+    foreach(plugin_path ${plugins_path})
+      file(APPEND ${CMAKE_CURRENT_SOURCE_DIR}/hooks/hook.dsv.in 
+      "prepend-non-duplicate;IGN_GAZEBO_SYSTEM_PLUGIN_PATH;@CMAKE_INSTALL_PREFIX@/${plugin_path}\n")
+    endforeach()
+    foreach(gui_plugin_path ${gui_plugins_path})
+      file(APPEND ${CMAKE_CURRENT_SOURCE_DIR}/hooks/hook.dsv.in 
+      "prepend-non-duplicate;IGN_GUI_PLUGIN_PATH;@CMAKE_INSTALL_PREFIX@/${gui_plugin_path}\n")
+    endforeach()
+    foreach(variable_export_path ${variable_export_paths})
+      string(REPLACE "," ";" variable_export_path ${variable_export_path})
+      file(APPEND ${CMAKE_CURRENT_SOURCE_DIR}/hooks/hook.dsv.in
+      "prepend-non-duplicate;${variable_export_path}\n")
+    endforeach()
+    file(WRITE ${CMAKE_CURRENT_SOURCE_DIR}/colcon.pkg "{\n  \"hooks\": [\"share/${CMAKE_PROJECT_NAME}/hooks/hook.dsv\"]\n}")
```

Comment:
I'm again not sure we want this here as the first run this will not work. We could alternatively just have these as part of the template you generate in https://github.com/gazebosim/gz_pkg_create

Reply:
I think we can leave it here and create it there as well because someone might not use that tool but use these macros